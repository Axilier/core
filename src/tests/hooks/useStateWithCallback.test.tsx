import React from 'react';
import { mount } from 'enzyme';
import { useStateWithCallback, useStepper } from '../../index';

// region demo
const Demo = ({ callback }: { callback: () => void }): JSX.Element => {
    const [state, setState] = useStateWithCallback(1, callback);
    return (
        <div id={'container'}>
            <div id={'state'}>{state}</div>
            <button
                type={'button'}
                id={'increment'}
                onClick={() => setState(state + 1)}
            >
                increment
            </button>
        </div>
    );
};
// endregion

const callback = jest.fn();
const demo = mount(<Demo callback={callback} />);

describe('when I render the demo in an application', () => {
    it('should render correctly', () => {
        expect(demo).toMatchSnapshot();
    });
    it('should set the initial state to 1', () => {
        expect(demo.find('#state').text()).toBe('1');
    });
    describe('when I click the button to change the state', () => {
        beforeAll(() => demo.find('#increment'));
        it('should call the callback', () => {
            expect(callback).toHaveBeenCalledWith(1);
        });
    });
});
