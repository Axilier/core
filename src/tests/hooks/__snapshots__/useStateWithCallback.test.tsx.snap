// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when I render the demo in an application should render correctly 1`] = `
<Demo
  callback={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    id="container"
  >
    <div
      id="state"
    >
      1
    </div>
    <button
      id="increment"
      onClick={[Function]}
      type="button"
    >
      increment
    </button>
  </div>
</Demo>
`;
